File: src/test/java/com/study/platform/support/fake/FakeTokenManager.java
package com.study.platform.support.fake;

import com.study.platform.global.auth.port.TokenManager;
import com.study.platform.global.exception.CustomException;
import com.study.platform.global.exception.ErrorCode;

import java.util.HashMap;
import java.util.Map;
import java.util.UUID;

public class FakeTokenManager implements TokenManager {

    private final Map<String, UUID> tokenStore = new HashMap<>();
    private final Map<UUID, String> invalidTokens = new HashMap<>();

    @Override
    public String generateAccessToken(UUID userId) {
        String token = "access-" + userId;
        tokenStore.put(token, userId);
        return token;
    }

    @Override
    public String generateRefreshToken(UUID userId) {
        String token = "refresh-" + userId;
        tokenStore.put(token, userId);
        return token;
    }

    @Override
    public void validateToken(String token) {
        if (!tokenStore.containsKey(token)) {
            throw new CustomException(ErrorCode.INVALID_TOKEN);
        }
    }

    @Override
    public UUID getUserIdFromToken(String token) {
        if (!tokenStore.containsKey(token)) {
            throw new CustomException(ErrorCode.INVALID_TOKEN);
        }
        return tokenStore.get(token);
    }
}

File: domain/src/testFixtures/kotlin/com/study/platform/support/fake/FakeTokenManager.kt
package com.study.platform.support.fake

import com.study.platform.global.auth.port.TokenManager
import com.study.platform.global.exception.CustomException
import com.study.platform.global.exception.ErrorCode
import java.util.UUID

class FakeTokenManager : TokenManager {

    private val tokenStore: MutableMap<String, UUID> = HashMap()

    override fun generateAccessToken(userId: UUID): String {
        val token = "access-$userId"
        tokenStore[token] = userId
        return token
    }

    override fun generateRefreshToken(userId: UUID): String {
        val token = "refresh-$userId"
        tokenStore[token] = userId
        return token
    }

    override fun validateToken(token: String) {
        if (!tokenStore.containsKey(token)) {
            throw CustomException(ErrorCode.INVALID_TOKEN)
        }
    }

    override fun getUserIdFromToken(token: String): UUID {
        if (!tokenStore.containsKey(token)) {
            throw CustomException(ErrorCode.INVALID_TOKEN)
        }
        return tokenStore[token]!!
    }
}
